cuckoo/root_node.py

```diff
@@ -138,50 +172,38 @@ def _build_request(self):
                 f"Request created. {query=}, variables={to_truncated_str(variables)}."
             )
 
-        return self._session.build_request(
+        return session.build_request(
             method="POST",
-            url=self._config["url"],
+            url=url,
             headers=self._config["headers"],
             json=json,
         )
 
     def _execute(self, stream=False):
-        request = self._build_request()
+        request = self._build_request(self.session)
         if self._logger:
             if stream:
                 self._logger.debug("Dispatching http streaming request.")
             else:
                 self._logger.debug("Dispatching http request.")
 
-        try:
-            for attempt in Retrying(**self._config["retry"]):
-                with attempt:
-                    self._response = self._session.send(request=request, stream=stream)
-                    self._response.raise_for_status()
-                    if not stream:
-                        self._process_response()
-        except RetryError as err:
-            raise HasuraServerError(repr(err))
-        finally:
-            if self._close_session and self._session is not None:
-                self._session.close()
+        for attempt in Retrying(**self._config["retry"]):
+            with attempt:
+                self._response = self.session.send(request=request, stream=stream)
+                self._response.raise_for_status()
+                if not stream:
+                    self._process_response()
 
     async def _execute_async(self):
-        request = self._build_request()
+        request = self._build_request(self.session_async)
         if self._logger:
             self._logger.debug("Dispatching asynchronous http request.")
 
-        try:
-            async for attempt in AsyncRetrying(**self._config["retry"]):
-                with attempt:
-                    self._response = await self._session_async.send(request=request)
-                    self._response.raise_for_status()
-                    self._process_response()
-        except RetryError as err:
-            raise HasuraServerError(repr(err))
```

Comment:
new: no re-raise of retry errors, as that hides the actual error. instead, it just bubbles up.